volatility/framework/symbols/windows/extensions/__init__.py

```diff
@@ -790,12 +793,29 @@ def get_sids(self) -> Iterable[str]:
 
 
     def privileges(self):
-        "Return a list of privileges for the current token object."
-        for priv_index in range(64):
-            yield (priv_index,
-                   bool(self.Privileges.Present & (2**priv_index)),
-                   bool(self.Privileges.Enabled & (2**priv_index)),
-                   bool(self.Privileges.EnabledByDefault & (2**priv_index)))
+        """Return a list of privileges for the current token object."""
+        
+        try:
+            for priv_index in range(64):
+                yield (priv_index,
+                       bool(self.Privileges.Present & (2**priv_index)),
+                       bool(self.Privileges.Enabled & (2**priv_index)),
+                       bool(self.Privileges.EnabledByDefault & (2**priv_index)))
+        except AttributeError: # Windows XP
+            layer_name = self.vol.layer_name
+            kvo = self._context.layers[layer_name].config["kernel_virtual_offset"]        
```

Comment:
We've let a few of these slip in, but if possible please use the public `self.context` to get the context, rather than referring to the private internal value `self._context`.  It's just in case in the future we want to calculate/change the context, and the context property would let us do that easily, whereas the private variable references would make it trickier for use to do (all highly unlikely, so definitely not a blocker, just a good practice thing).  5:)

Reply:
only the _context exists

Reply:
Ah, you're quite right, my mistake.  I thought we exposed it as a property.  Ignore this then!  5:)